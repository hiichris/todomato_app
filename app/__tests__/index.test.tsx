import React from "react";
import renderer from "react-test-renderer";
import { waitFor } from "@testing-library/react-native";
import { NativeDatabase } from "expo-sqlite";

import App from "../___index";

class ErrorBoundary extends React.Component {
  // Suggested by React Native documentation to handle errors in the app
  state = { hasError: false };

  static getDerivedStateFromError(error) {
    return { hasError: true };
  }

  componentDidCatch(error, errorInfo) {
    // TODO: To research on how to resolve: 
    // TypeError: _ExpoSQLiteNext.default.NativeDatabase is not a constructor
    // Therefore commenting out the console.error

    // console.error(error, errorInfo);
  }

  render() {
    if (this.state.hasError) {
      // Render fallback UI here
      return <div>Error occurred.</div>;
    }

    return this.props.children;
  }
}

describe("<App />", () => {
  it("renders correctly", async () => {
    const tree = renderer
      .create(
        <ErrorBoundary>
          <App />
        </ErrorBoundary>
      )
      .toJSON();
    await waitFor(() => expect(tree).toMatchSnapshot());
  });
});
