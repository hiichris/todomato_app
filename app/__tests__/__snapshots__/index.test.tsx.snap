// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> has 1 child 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
/>
`;
